app/models/test.rb

```diff
@@ -6,5 +6,17 @@ class Test < ActiveRecord::Base
 
   accepts_nested_attributes_for :answers
 
+  after_create :count_result
+
   resourcify#rolify
+
+  class_attribute :current_user
+
+  def count_result
+    true_answers = Answer.current_test(Test.last.id).current_student(current_user.id).with_true.joins(:answer_setting).where( answer_settings: {rigth: 'true'} )
+    # true_answers = Answer.all
+    all_questions = Test.last.test_setting.questions
+    result = true_answers.count / all_questions.count * 100
```

Comment:
convert one variable to float, like Float(true_answers.count)

because 1/2 #=> 0
1.0/2 #=> 0.5